features/steps/steps.go

```diff
@@ -33,6 +33,7 @@ func (c *IdentityComponent) RegisterSteps(ctx *godog.ScenarioContext) {
 	ctx.Step(`^an internal server error is returned from Cognito$`, c.anInternalServerErrorIsReturnedFromCognito)
 	ctx.Step(`^group "([^"]*)" and description "([^"]*)" exists in the database$`, c.groupAndDescriptionExistsInTheDatabase)
 	ctx.Step(`^group "([^"]*)" exists in the database$`, c.groupExistsInTheDatabase)
+	ctx.Step(`^group "([^"]*)" exists in the database to sort$`, c.groupExistsInTheDatabaseSort)
```

Comment:
do we need this differentiation with line 35?

Reply:
> do we need this differentiation with line 35?

Kind of hard to explain without a demonstration, but the original groupExistsInTheDatabase function doesn't use the group value we need for sorting. It just creates a Group, whereas we need *cognitoidentityprovider.ListGroupsOutput. They both exist within the same struct but are otherwise unrelated, so I couldn't figure out any way to get around this other than creating a whole new set of functions and a new step. The existing functions couldn't really be modified from what I could see